TLM/TLM/State/OptionsTabs/MaintenanceTab_AllowDespawnGroup.cs

```diff
@@ -0,0 +1,159 @@
+            new ("AllowDespawnCargoPlane", Options.PersistTo.Global) {
+                Label = "AllowDespawn.Checkbox:Cargo Planes",
+                Handler = (bool newValue) => OnChange(ExtVehicleType.CargoPlane, newValue),
+                Value = GlobalConfig.Instance.Gameplay.AllowedDespawnVehicleTypes.IsFlagSet(ExtVehicleType.CargoPlane)
+            };
+        */
+
+        public static CheckboxOption AllowDespawnCargoTrains =
+            new ("AllowDespawnCargoTrain", Options.PersistTo.Global) {
+                Label = "AllowDespawn.Checkbox:Cargo Trains",
+                Handler = (bool newValue) => OnChange(ExtVehicleType.CargoTrain, newValue),
+                Value = GlobalConfig.Instance.Gameplay.AllowedDespawnVehicleTypes.IsFlagSet(ExtVehicleType.CargoTrain)
+            };
+        public static CheckboxOption AllowDespawnCargoTrucks =
+            new ("AllowDespawnPassengerTruck", Options.PersistTo.Global) {
+                Label = "AllowDespawn.Checkbox:Cargo Trucks",
+                Handler = (bool newValue) => OnChange(ExtVehicleType.CargoTruck, newValue),
+                Value = GlobalConfig.Instance.Gameplay.AllowedDespawnVehicleTypes.IsFlagSet(ExtVehicleType.CargoTruck)
+            };
+        public static CheckboxOption AllowDespawnService =
+            new ("AllowDespawnService", Options.PersistTo.Global) {
+                Label = "AllowDespawn.Checkbox:Service vehicles",
+                Handler = (bool newValue) => OnChange(ExtVehicleType.Service, newValue),
+                Value = GlobalConfig.Instance.Gameplay.AllowedDespawnVehicleTypes.IsFlagSet(ExtVehicleType.Service)
+            };
+
+        private static readonly Dictionary<CheckboxOption, ExtVehicleType> AllowDespawns = new ()
+        {
+            { AllowDespawnPassengerCars, ExtVehicleType.PassengerCar },
+            { AllowDespawnBuses, ExtVehicleType.Bus },
+            { AllowDespawnTaxis, ExtVehicleType.Taxi },
+            { AllowDespawnTrams, ExtVehicleType.Tram },
+            { AllowDespawnTrolleybuses, ExtVehicleType.Trolleybus },
+            /*TODO #1368 - not supported yet
+            { AllowDespawnPassengerPlanes, ExtVehicleType.PassengerPlane },
+            */
+            { AllowDespawnPassengerTrains, ExtVehicleType.PassengerTrain },
+            /*TODO #1368 - not supported yet
+             { AllowDespawnCargoPlanes, ExtVehicleType.CargoPlane },
+             */
+            { AllowDespawnCargoTrains, ExtVehicleType.CargoTrain },
+            { AllowDespawnCargoTrucks, ExtVehicleType.CargoTruck },
+            { AllowDespawnService, ExtVehicleType.Service },
+        };
+
+        static MaintenanceTab_AllowDespawnGroup() {
+            // attach observer, update checkboxes with new values
+            GlobalConfigObserver configObserver = TMPELifecycle.Instance.gameObject.AddComponent<GlobalConfigObserver>();
+            configObserver.OnUpdateObservers += (config) => OnUpdateCheckboxes(config);
```

Comment:
What can break here? 
Reading `GlobalConfig` in the `GlobalConfigObserver`? If so then the mod will be broken anyways as it's not the first access to it and won't work properly without GlobalConfig.

Reply:
IMO even if nothing can break, we should still wrap body of static constructors in `try..catch` just as standard practice. Maybe I just have PTSD from having 140+ mods disabled due to issues in static constructors when doing some of the earlier work on `CheckboxOption` which caused exception in the constructor = boom. Some mods, ambient sound tuner in particular, need re-enabling in specific order to avoid settings loss.